--- a/bfd/configure
+++ b/bfd/configure
@@ -3183,7 +3183,7 @@ fi
 
 
 # Define the identity of the package.
- PACKAGE='bfd'
+ PACKAGE=cygwin-bfd
  VERSION='2.42'
 
 
--- a/binutils/configure
+++ b/binutils/configure
@@ -3080,7 +3080,7 @@ fi
 
 
 # Define the identity of the package.
- PACKAGE='binutils'
+ PACKAGE=cygwin-binutils
  VERSION='2.42'
 
 
--- a/gas/configure
+++ b/gas/configure
@@ -2989,7 +2989,7 @@ fi
 
 
 # Define the identity of the package.
- PACKAGE='gas'
+ PACKAGE=cygwin-gas
  VERSION='2.42'
 
 
--- a/gprof/configure
+++ b/gprof/configure
@@ -2863,7 +2863,7 @@ fi
 
 
 # Define the identity of the package.
- PACKAGE='gprof'
+ PACKAGE=cygwin-gprof
  VERSION='2.42'
 
 
--- a/ld/configure
+++ b/ld/configure
@@ -3358,7 +3358,7 @@ fi
 
 
 # Define the identity of the package.
- PACKAGE='ld'
+ PACKAGE=cygwin-ld
  VERSION='2.42'
 
 
--- a/opcodes/configure
+++ b/opcodes/configure
@@ -3117,7 +3117,7 @@ fi
 
 
 # Define the identity of the package.
- PACKAGE='opcodes'
+ PACKAGE=cygwin-opcodes
  VERSION='2.42'
 
 
